File: publish_extension.py
import json
import os
import subprocess

def update_version(package_json_path):
    with open(package_json_path, 'r') as f:
        data = json.load(f)

    version = data['version']
    major, minor, patch = [int(x) for x in version.split('.')]
    
    # Increment the patch version
    patch += 1

    new_version = f"{major}.{minor}.{patch}"
    data['version'] = new_version

    with open(package_json_path, 'w') as f:
        json.dump(data, f, indent=2)

    return new_version

def main():
    package_json_path = "package.json"
    token_path = "C:\\PosetMage\\Package\\_token\\azure_token"

    new_version = update_version(package_json_path)

    # Run git commands
    subprocess.run(["git", "add", "."])
    subprocess.run(["git", "commit", "-m", f"Update to version {new_version}"])
    subprocess.run(["git", "push"])

    # Package the extension
    subprocess.run(["vsce", "package"])

    # Publish the extension
    with open(token_path, "r") as f:
        token = f.read().strip()

    subprocess.run(["vsce", "publish", "--pat", token])

if __name__ == "__main__":
    main()

File: publish_extension.js
const fs = require('fs');
const { execSync } = require('child_process');

function updateVersion(packageJsonPath) {
  const data = fs.readFileSync(packageJsonPath, 'utf8');
  const { version } = JSON.parse(data);
  const [major, minor, patch] = version.split('.').map(Number);
  
  // Increment the patch version
  const newPatch = patch + 1;

  const newVersion = `${major}.${minor}.${newPatch}`;
  const newData = { ...JSON.parse(data), version: newVersion };
  fs.writeFileSync(packageJsonPath, JSON.stringify(newData, null, 2));

  return newVersion;
}

function main() {
  const packageJsonPath = 'package.json';
  const tokenPath = 'C:\\PosetMage\\Package\\_token\\azure_token';

  const newVersion = updateVersion(packageJsonPath);

  // Run git commands
  execSync(`git add .`);
  execSync(`git commit -m "Update to version ${newVersion}"`);
  execSync(`git push`);

  // Package the extension
  execSync(`vsce package`);

  // Publish the extension
  const token = fs.readFileSync(tokenPath, 'utf8').trim();
  execSync(`vsce publish --pat ${token}`);
}

main();